src/pudl/output/pudltabl.py

```diff
@@ -249,29 +261,113 @@ def _register_output_methods(self: Self):
                     "explicitly defined class method. One of these should be deleted."
                 )
 
+        if self.table_source == "parquet":
+            self.__dict__["get_table"] = self._get_table_from_parquet
+        else:
+            self.__dict__["get_table"] = self._get_table_from_sqlite
+
         for table_name, method_name in table_method_map_any_freq.items():
             # Create method called table_name that will read the asset from DB
             self.__dict__[method_name] = partial(
-                self._get_table_from_db,
+                self.get_table,
                 table_name=table_name,
                 allowed_freqs=[None, "YS", "MS"],
             )
 
         for table_name, method_name in table_method_map_any_agg.items():
             self.__dict__[method_name] = partial(
-                self._get_table_from_db,
+                self.get_table,
                 table_name=table_name,
                 allowed_freqs=["YS", "MS"],
             )
 
         for table_name, method_name in table_method_map_yearly_only.items():
             self.__dict__[method_name] = partial(
-                self._get_table_from_db,
+                self.get_table,
                 table_name=table_name,
                 allowed_freqs=["YS"],
             )
 
-    def _get_table_from_db(
+    def _get_table_from_parquet(  # noqa: C901
```

Comment:
This method mirrors both `_get_table_from_db()` and the `ParquetIOManager` to try and speed up & reduce the memory intensity of reading PUDL Tables in our CI & testing.